docs/reference/api.md

```diff
@@ -0,0 +1,77 @@
+# API
+
+Pebble exposes an API to allow remote clients to interact with the daemon. The API can start and stop services, add configuration layers to the plan, and so on.
+
+## Accessing the API
+
+The API uses HTTP over a Unix socket, with access to the API controlled by user ID. If `pebble run` is started with the `--http <address>` option, Pebble exposes a limited set of open-access API endpoints using the given TCP address. See {ref}`api-access-levels`.
+
+See below for some examples of how to use the API. For more examples, see "How to use the Pebble API". <!-- [David] Link to the how-to guide -->
+
+There's a Go client for the API. See the [Go client documentation](https://pkg.go.dev/github.com/canonical/pebble/client) and the examples below.
+
+We try to never change the underlying HTTP API in a backwards-incompatible way. However, in rare cases we may change the Go client in a backwards-incompatible way.
```

Comment:
This paragraph is kind of stuck in the middle of mentioning the Go client and the Python client. How about we move it up.

In fact, instead of having these paragraphs here, why don't we make the structure like this:

```
# Accessing the API

## Using curl 

Introductory paragraph and a couple of examples (one GET, one POST).

## Using the Go client

Introductory paragraph and one example (one is probably enough here, just the one from https://pkg.go.dev/github.com/canonical/pebble/client).

## Using the Python client

Paragraph mentioning and linking to the docs (can we link to https://ops.readthedocs.io/en/latest/pebble.html to be more specific?). Maybe a 3-line example like:

     import ops
     client = ops.pebble.Client('/path/to/.pebble.socket')
     client.start_services(['svc1', 'svc2'])
```